crypto/src/transaction.rs

```diff
@@ -15,17 +22,23 @@ impl TransactionBody {
     }
 }
 
-pub struct Transaction {}
+// temp: remove dead code when Transaction fields are read
+#[allow(dead_code)]
+pub struct Transaction {
+    transaction_body: TransactionBody,
+    binding_sig: Signature<Binding>,
+}
 
 impl Transaction {
     /// Start building a transaction relative to a given [`merkle::Root`].
     pub fn build_with_root(merkle_root: merkle::Root) -> TransactionBuilder {
         TransactionBuilder {
-            spends: Vec::<Spend>::new(),
-            outputs: Vec::<Output>::new(),
+            actions: Vec::<Action>::new(),
```

Comment:
This can just be `Vec::new()`; the inner type is already constrained by the `TransactionBuilder` definition.